golang/citadel.go

```diff
@@ -0,0 +1,153 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"io/ioutil"
+	"encoding/json"
+	"os"
+	"bytes"
+)
+
+type PublicTokenRequest struct {
```

Comment:
It will be good to add a comment to each function and structure in form:
```
// PublicTokenRequest is ...
type PublicTokenRequest struct {
```
https://golang.org/doc/effective_go.html#commentary